provider/datagen/src/cldr/transform/time_zones/mod.rs

```diff
@@ -61,11 +68,39 @@ macro_rules! impl_resource_provider {
                             .time_zone_names))
                     };
 
+                    if self.bcp47_tzid_data.read().unwrap().len() == 0 {
+                        let bcp47_time_zone_path = self.bcp47_tzid_path.join("timezone.json");
+
+                        let resource: cldr_serde::time_zones::bcp47_tzid::Resource =
+                            serde_json::from_reader(open_reader(&bcp47_time_zone_path)?)
+                                .map_err(|e| Error::Json(e, Some(bcp47_time_zone_path)))?;
+                         let r = resource
+                            .keyword
+                            .u
+                            .time_zones
+                            .values;
+
+                        let mut data_guard = self.bcp47_tzid_data.write().unwrap();
+                        for (bcp47_tzid, bcp47_tzid_data) in r.iter() {
+                            if let Some(alias) = &bcp47_tzid_data.alias {
+                                for data_value in alias.split(" ") {
+                                    data_guard.insert(data_value.to_string(), bcp47_tzid.to_string());
+                                }
+                            }
+                        }
+                    }
+                    let bcp47_tzids = self.bcp47_tzid_data.read().unwrap().clone();
+
+                    let cldr_time_zones_data = CldrTimeZonesData {
+                        time_zone_names: time_zone_names.clone(),
```

Comment:
@sffc @Manishearth `time_zone_names` is a reference (not 'static or owned). Do you know if there is a better way to handle it instead of cloning? 

If I make struct `CldrTimeZonesData` take reference `time_zone_names`, it will raise error  '`self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement'.

Note that `cldr_time_zones_data` will be passed in `Yokeable::from` in line 103.

Reply:
We don't care too much about clones in datagen, this is fine.

Reply:
I suggest changing `CldrTimeZonesData` to be

```rust
pub struct CldrTimeZonesData<'a> {
    pub time_zone_names: &'a TimeZoneNames,
    pub bcp47_tzids: &'a LiteMap<String, String>,
}
```

and then you can save references.

> We don't care too much about clones in datagen, this is fine.

This is generally true, but this is effectively cloning a large object in a tight loop, which would be nice to avoid, since we have an easy way to avoid it.

Reply:
Also, consider moving the struct to be inside the `transform` tree, rather than `cldr_serde`, since it isn't constructed or used anywhere else.

Reply:
> I suggest changing `CldrTimeZonesData` to be
> 
> ```rust
> pub struct CldrTimeZonesData<'a> {
>     pub time_zone_names: &'a TimeZoneNames,
>     pub bcp47_tzids: &'a LiteMap<String, String>,
> }
> ```
> 
> and then you can save references.

Sam tried this and ran into issues that Yoke requires a static lifetime. 

The error that Sam mentioned above: 
```
`self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement.
```

Essentially, we can't guarantee that the lifetime of `Self`, to which the reference to `TimeZoneNames` belongs, would necessarily outlive the data payload that we create via:

```rust
Some(DataPayload::from_owned(<$marker as DataMarker>::Yokeable::from(cldr_time_zones_data))
```

I told Sam to not worry about the clones and that we could tackle this as a separate efficiency problem if it causes performance issues. 

The lifetime reasoning from the compiler seems sound to me. I think it is correct that `Self` would not necessarily outlive the data payload, but I'm less familiar with the internals of Yoke. 

@sffc if you have the solution for this, then we should apply it, but I didn't see one myself.

Reply:
CldrTimeZonesData is never itself converted to a Yokeable. The From impl should clone the data, but only the small part of the data it needs.

Reply:
It just seems to be that the data contained within `CldrTimeZonesData` needs to be owned, i.e. copied before the `From` impl is even invoked, otherwise it doesn't meet Yoke's `'static` lifetime requirements. 

It would be great to be able to pass a reference into the `From` impl, but it doesn't seem to be compatible with Yoke.

Am I misunderstanding something?

Reply:
There are multiple things going on here:

```
Some(
    DataPayload::from_owned(
        <$marker as DataMarker>::Yokeable::from(
            cldr_time_zones_data
)))
```

`<$marker as DataMarker>::Yokeable::from` expands to something like `ExemplarCitiesV1<'static>::from`, and it can invoke a `From` impl such as

```rust
impl From<&CldrTimeZonesData<'_>> for ExemplarCitiesV1<'static>
```

Then, you call `DataPayload::from_owned(...)` which is the thing that requires the `'static`, but that's OK, because we have it since it is the return value of the `impl From`.

Reply:
So @samchen61661 based on what Shane just said it looks like you may need to make the `From` impls take a reference to `CldrTimeZoneNames` instead of by value.

That is,

```rust
impl From<&CldrTimeZonesData<'_>> for ExemplarCitiesV1<'static>
```

instead of what it is currently:

```rust
impl From<CldrTimeZonesData<'_>> for ExemplarCitiesV1<'_>
```

Then hopefully things will work with references.

Reply:
What you currently have is

```rust
impl From<CldrTimeZonesData> for TimeZoneFormatsV1<'_> {
```

You need to make it

```rust
impl From<&CldrTimeZonesData<'_>> for TimeZoneFormatsV1<'static> {
```

It doesn't actually matter whether `CldrTimeZonesData<>` is by value or reference. What matters is that the lifetime parameter on `CldrTimeZonesData<>` is not constrained to be the same as the lifetime parameter on `TimeZoneFormatsV1<>`.

Reply:
great! It is working now!